lib/broccoli/amd-excluder.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const Funnel = require('broccoli-funnel');
+const walkSync = require('walk-sync');
+
+class AmdExcluder extends Funnel {
```

Comment:
can you use https://github.com/broccolijs/broccoli-test-helper to write a few tests for this class?

Reply:
a little bit of documentation on this class might also be nice. e.g. answering the question that @rtablada raised about this being specialized to detect AMD modules compiled by Babel (5? 6? 7?)

Reply:
added docs